From: Colin Watson <cjwatson@debian.org>
Date: Tue, 29 Oct 2024 10:44:19 +0000
Subject: test: replace more deprecated assertEquals

`TestCase.assertEquals` was deprecated and removed in Python 3.12.

Similar to #350, but for code that was added later.

Origin: other, https://github.com/ansible-collections/community.grafana/pull/406
Last-Update: 2024-10-29
---
 .../community/grafana/changelogs/fragments/406-python3.12.yml         | 3 +++
 .../unit/modules/grafana/grafana_silence/test_grafana_silence.py      | 4 ++--
 2 files changed, 5 insertions(+), 2 deletions(-)
 create mode 100644 ansible_collections/community/grafana/changelogs/fragments/406-python3.12.yml

diff --git a/ansible_collections/community/grafana/changelogs/fragments/406-python3.12.yml b/ansible_collections/community/grafana/changelogs/fragments/406-python3.12.yml
new file mode 100644
index 0000000..376cc7a
--- /dev/null
+++ b/ansible_collections/community/grafana/changelogs/fragments/406-python3.12.yml
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - "test: replace more deprecated `TestCase.assertEquals` to support Python 3.12"
diff --git a/ansible_collections/community/grafana/tests/unit/modules/grafana/grafana_silence/test_grafana_silence.py b/ansible_collections/community/grafana/tests/unit/modules/grafana/grafana_silence/test_grafana_silence.py
index 96522c2..74d7b4b 100644
--- a/ansible_collections/community/grafana/tests/unit/modules/grafana/grafana_silence/test_grafana_silence.py
+++ b/ansible_collections/community/grafana/tests/unit/modules/grafana/grafana_silence/test_grafana_silence.py
@@ -162,7 +162,7 @@ class GrafanaSilenceTest(TestCase):
             },
             method="POST",
         )
-        self.assertEquals(result, {"silenceID": "470b7116-8f06-4bb6-9e6c-6258aa92218e"})
+        self.assertEqual(result, {"silenceID": "470b7116-8f06-4bb6-9e6c-6258aa92218e"})
 
     @patch(
         "ansible_collections.community.grafana.plugins.modules.grafana_silence.GrafanaSilenceInterface.get_version"
@@ -208,4 +208,4 @@ class GrafanaSilenceTest(TestCase):
             },
             method="DELETE",
         )
-        self.assertEquals(result, {"message": "silence deleted"})
+        self.assertEqual(result, {"message": "silence deleted"})
